### ISSUE TITLE
LogCollectBrowserConsole serialization issue

### ISSUE THREAD

**Issue by ming-codes** [archfz/cypress-terminal-report#192]
```
tests?p=cypress/support/e2e.ts:40561 TypeError: Cannot convert object to primitive value
    at processArg (tests?p=cypress/support/e2e.ts:40546:36)
    at Array.map (<anonymous>)
    at appWindow.console.<computed> [as warn] (tests?p=cypress/support/e2e.ts:40560:53)
```

The error is stem from this line, where it caught an error but try to serialize it again by attempting to concatenate it.

https://github.com/archfz/cypress-terminal-report/blob/72123c8547916bf007921671853b92351093c6b4/src/collector/LogCollectBrowserConsole.js#L39

The other issue is already described by #191, but it is actually more generic. `JSON.stringify` can throw simply on object that's too big.

**Comment by archfz** [comment]
I was thinking on using https://www.npmjs.com/package/safe-json-stringify but maybw this case would not be covered still.

**Comment by ming-codes** [comment]
You think we can add an option for maximum depth to serialize? We can include a good default.

**Comment by archfz** [comment]
Should be fixed in 5.3.0
